apps/ckit-app/src/hooks/useSendTransferTx.tsx

```diff
@@ -1,76 +1,52 @@
   const ckitProvider = CkitProviderContainer.useContainer();
-  const [localConfig] = useConfigStorage();
+  const { cacheTx } = useUnipass();
+  const { mutateAsync: sendTransaction } = useSendTransaction();
 
-  return useMutation(
-    ['sendTransferTx'],
-    async (input: SendTransferTxInput) => {
-      if (!currentWallet?.signer) throw new Error('exception: signed undifined');
-      if (!ckitProvider) throw new Error('exception: ckitProvider undifined');
-      let txToSend: unknown;
+  return useMutation(['sendTransferTx'], async (input: SendTransferTxInput) => {
+    if (!currentWallet?.signer) throw new Error('exception: signed undifined');
+    if (!ckitProvider) throw new Error('exception: ckitProvider undifined');
+    let txToSend: Transaction;
 
-      if (input.script) {
-        const txBuilder = new AcpTransferSudtBuilder(
-          {
-            recipient: input.recipient,
-            sudt: input.script,
-            amount: input.amount,
-          },
-          ckitProvider,
-          currentWallet.signer,
-        );
-        txToSend = await txBuilder.build();
-      } else {
-        const txBuilder = new TransferCkbBuilder(
-          { recipients: [{ recipient: input.recipient, amount: input.amount, capacityPolicy: 'createAcp' }] },
-          ckitProvider,
-          currentWallet.signer,
-        );
-        txToSend = await txBuilder.build();
-      }
-      const txHash = await ckitProvider.sendTransaction(await currentWallet.signer.seal(txToSend));
-      return { txHash: txHash };
-    },
-    {
-      onSuccess({ txHash }) {
-        const href = localConfig.nervosExploreTxUrlPrefix + txHash;
-        Modal.success({
-          title: 'Tx sent',
-          content: (
-            <p>
-              The transaction was sent, check it in&nbsp;
-              <a href={href} target="_blank" rel="noreferrer">
-                explorer
-              </a>
-              <details>
-                <summary>transaction id</summary>
-                {txHash}
-              </details>
-            </p>
-          ),
-        });
-      },
-      onError(error) {
-        const errorMsg: string = hasProp(error, 'message') ? String(error.message) : 'Unknown error';
-        Modal.error({ title: 'Tx failed', content: errorMsg, width: 360 });
-      },
-    },
-  );
+    if (input.script) {
+      const txBuilder = new AcpTransferSudtBuilder(
+        {
+          recipient: input.recipient,
+          sudt: input.script,
+          amount: input.amount,
+        },
+        ckitProvider,
+        currentWallet.signer,
+      );
+      txToSend = await txBuilder.build();
+    } else {
+      const txBuilder = new TransferCkbBuilder(
+        { recipients: [{ recipient: input.recipient, amount: input.amount, capacityPolicy: 'createAcp' }] },
+        ckitProvider,
+        currentWallet.signer,
+      );
+      txToSend = await txBuilder.build();
+    }
+    if (currentWallet.descriptor.name === 'UniPass') {
```

Comment:
the UniPass check can be moved to `sendTransaction`, because there may be scenarios where UniPass will be used to send non-transfer transactions later

Reply:
fixed